components/post_edit_history/restore_post_modal.tsx

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {memo, useRef} from 'react';
+import {Modal} from 'react-bootstrap';
+import {FormattedMessage} from 'react-intl';
+
+import {useSelector} from 'react-redux';
+
+import {Post} from '@mattermost/types/posts';
+import {isModalOpen} from 'selectors/views/modals';
+import {GlobalState} from 'types/store';
+import {ModalIdentifiers} from 'utils/constants';
+
+type Props = {
+    post: Post;
+    actions: {
+        handleRestore: (post: Post) => void;
+        closeModal: (modalId: string) => void;
+    };
+}
+
+const RestorePostModal = ({post, actions}: Props) => {
+    const restorePostBtn = useRef<HTMLButtonElement>(null);
+    const show = useSelector((state: GlobalState) => isModalOpen(state, ModalIdentifiers.RESTORE_POST_MODAL));
+
+    const onHide = () => actions.closeModal(ModalIdentifiers.RESTORE_POST_MODAL);
```

Comment:
based on @hmhealey comment ealier, you might be able to close the modal, you might be able to use props.onExited() here also.